File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int,int> mp;
        int n=nums.size();
        for(int i=0;i<n;i++){
            int m=target-nums[i];
            if(mp.find(m) != mp.end()){
                return {{mp[m],i}};
            }
            mp[nums[i]]=i;
        }

        return {};
    }
};

File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        mp={}
        n=len(nums)
        for i in range(0,n):
            x=target-nums[i]
            val=mp.get(x, None)
            if val is not None:
                return {mp[x],i}
            mp[nums[i]]=i

        return {}
        